test/functional/mempool_unbroadcast.py

```diff
@@ -0,0 +1,121 @@
+
+        disconnect_nodes(node, 1)
+
+        self.log.info("Generate transactions that only node 0 knows about")
+        # generate a wallet txn
+        addr = node.getnewaddress()
+        wallet_tx_hsh = node.sendtoaddress(addr, 0.0001)
+
+        # generate a txn using sendrawtransaction
+        us0 = utxos.pop()
+        inputs = [{ "txid" : us0["txid"], "vout" : us0["vout"]}]
+        outputs = {addr: 0.0001}
+        tx = node.createrawtransaction(inputs, outputs)
+        node.settxfee(min_relay_fee)
+        txF = node.fundrawtransaction(tx)
+        txFS = node.signrawtransactionwithwallet(txF['hex'])
+        rpc_tx_hsh = node.sendrawtransaction(txFS['hex'])  # txhsh in hex
+
+        # check that second node doesn't have these two txns
+        mempool = self.nodes[1].getrawmempool()
+        assert(rpc_tx_hsh not in mempool)
+        assert(wallet_tx_hsh not in mempool)
+
+        self.log.info("Reconnect nodes & check if they are sent to node 1")
+        connect_nodes(node, 1)
+
+        # fast forward into the future & ensure that the second node has the txns
+        node.mockscheduler(15*60) # 15 min in seconds
+        wait_until(lambda: len(self.nodes[1].getrawmempool()) == 2, timeout=30)
+        mempool = self.nodes[1].getrawmempool()
+        assert(rpc_tx_hsh in mempool)
+        assert(wallet_tx_hsh in mempool)
+
+        self.log.info("Add another connection & ensure transactions aren't broadcast again")
+
+        conn = node.add_p2p_connection(P2PTxInvStore())
+        node.mockscheduler(15*60)
+        time.sleep(5)
+        assert_equal(len(conn.get_invs()), 0)
+
+    def test_txn_removal(self):
+        self.log.info("Test that transactions removed from mempool are removed from unbroadcast set")
+        node = self.nodes[0]
+        disconnect_nodes(node, 1)
+
+        min_relay_fee = node.getnetworkinfo()["relayfee"]
+        utxos = create_confirmed_utxos(min_relay_fee, node, 10)
+
+        # create a transaction & submit to node
+        # since the node doesn't have any connections, it will not receive
+        # any GETDATAs & thus the transaction will remain in the unbroadcast set.
+        utxo = utxos.pop()
+        outputs = { node.getnewaddress() : 0.0001 }
+        inputs = [{'txid': utxo['txid'], 'vout': utxo['vout']}]
+        raw_tx_hex = node.createrawtransaction(inputs, outputs)
+        signed_tx = node.signrawtransactionwithwallet(raw_tx_hex)
+        txhsh = node.sendrawtransaction(hexstring=signed_tx['hex'], maxfeerate=0)
+
+        # mine a block with that transaction
+        block = create_block(int(node.getbestblockhash(), 16), create_coinbase(node.getblockcount() + 1))
+        tx = CTransaction()
+        tx.deserialize(BytesIO(hex_str_to_bytes(signed_tx['hex'])))
+        block.vtx.append(tx)
+        block.rehash()
+        block.hashMerkleRoot = block.calc_merkle_root()
+        block.solve()
+        node.submitblock(ToHex(block))
```

Comment:
in commit f40f687:

Does the block need to be created manually? What about a simple two-liner?

```
self.nodes[0].sendtoaddress()
self.nodes[0].generate(1)
```

Alternatively, if you want to test both wallet txs and rpc txs, you could factor out the tx-creation code from the previous test case into a new helper function and call it here.

Reply:
ahahhahaha omg I'm glad its been a while since I wrote this test & I don't remember how long I spent figuring out how to manually make the txn and block 😝

two-liner is much cleaner. I feel comfortable with the coverage of wallet & rpc from previous test.

thanks!